### PR TITLE
Add revisionHistoryLimit to VictoriaMetrics resources which creates Deployment

### PR DESCRIPTION
**Pull request by gidesh**
We have been doing some work to try to clean up unwanted resources in our cluster. Out of them is ReplicaSet created by Deployment.

So for VMAgent and VMAlerts we are trying to reduce the revisionHistoryLimit from the default 10 set by k8s to 3. However, the vmagent or vmalert spec does not extend this feature to update this limit. We tried to put a mutation policy in place to change it in the deployment created by VMAgent resource; however, the operator put it back to 10.

Fixes https://github.com/VictoriaMetrics/operator/issues/824

### CODE REVIEW

**Comment by gidesh** [comment]
The failing test might not have anything to do with the changes in the PR. It is failing at 
```
make[1]: Leaving directory '/home/runner/work/operator/operator'
make: *** No rule to make target 'docker-operator-crosscompile'.  Stop.
Error: Process completed with exit code 2.
```

File: e2e/vmcluster_test.go [hunk 1]
```diff
@@ -137,6 +137,28 @@ var _ = Describe("e2e vmcluster", func() {
 				Eventually(func() string {
 					return expectPodCount(k8sClient, 2, namespace, vmCluster.VMInsertSelectorLabels())
 				}, 70, 1).Should(BeEmpty())
+
+				By("vmInsert revisionHistoryLimit update to 3")
```
Comment (Haleygo) [inline e2e/vmcluster_test.go]: minor typo here)
"vmInsert revisionHistoryLimit update to 3" -> "vminsert revisionHistoryLimit update to 2"

**Review by Haleygo — COMMENTED** [review]
LGTM!
Could you help adding a feature description to the [changelog](https://github.com/VictoriaMetrics/operator/blob/master/docs/CHANGELOG.md) as well.

**Comment by Haleygo** [comment]
> The failing test might not have anything to do with the changes in the PR. It is failing at
> 
> ```
> make[1]: Leaving directory '/home/runner/work/operator/operator'
> make: *** No rule to make target 'docker-operator-crosscompile'.  Stop.
> Error: Process completed with exit code 2.
> ```

Yeah, don't mind about that, should be fixed in https://github.com/VictoriaMetrics/operator/pull/838.

File: e2e/vmcluster_test.go (hunk 1 above)
Comment (gidesh) [reply e2e/vmcluster_test.go]: Done

**Comment by gidesh** [comment]
> LGTM! Could you help adding a feature description to the [changelog](https://github.com/VictoriaMetrics/operator/blob/master/docs/CHANGELOG.md) as well?

Done

**Review by Haleygo — APPROVED** [review]
LGTM!

**Pull request merged**